daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/traversable/TypeTraversable.scala

```diff
@@ -15,6 +15,8 @@ private[validation] object TypeTraversable {
 
   private[validation] def foreach[U](typ: Type, f: Type => U): Unit =
```

Comment:
This 2 changes below, follow the current philosophy of the TypeTraversable iterator utility.
Without those changes, `Recursion` check will broke.

Reply:
As above

Reply:
What will break? I don't understand

Reply:
@remyhaemmerle-da is referring to `modRefsInType` in `com.digitalasset.daml.lf.validation.Recursion` which currently checks for `TTyCon` and therefore will miss module references in a type synonym name. That said, I’m not sure I agree that the suggestions make sense here. Pretending that a type synonym is a type con is really confusing. I think it makes much more sense to extend `modRefsInType` to handle type synonyms.

Reply:
Right. I see that now. 
And I agree a fix to `modRefsInType` would be clearer.

Reply:
Ok for the fix of  `modRefsInType`

Reply:
done